x/btcstaking/keeper/msg_server.go

```diff
@@ -353,21 +353,18 @@ func (ms msgServer) AddCovenantSigs(goCtx context.Context, req *types.MsgAddCove
 
 	if btcDel.IsSignedByCovMember(req.Pk) && btcDel.BtcUndelegation.IsSignedByCovMember(req.Pk) {
 		ms.Logger(ctx).Debug("Received duplicated covenant signature", "covenant pk", req.Pk.MarshalHex())
-		return &types.MsgAddCovenantSigsResponse{}, nil
+		// return error if the covenant signature is already submitted
+		// this is to secure the tx refunding against duplicated messages
+		return nil, types.ErrDuplicatedCovenantSig
 	}
 
-	if btcDel.HasCovenantQuorums(params.CovenantQuorum) {
-		ms.Logger(ctx).Debug("Received covenant signature after achieving quorum", "covenant pk", req.Pk.MarshalHex())
-		return &types.MsgAddCovenantSigsResponse{}, nil
-	}
-
-	// ensure BTC delegation is still pending, i.e., not expired
+	// ensure BTC delegation is still pending, i.e., not unbonded
 	btcTipHeight := ms.btclcKeeper.GetTipInfo(ctx).Height
 	wValue := ms.btccKeeper.GetParams(ctx).CheckpointFinalizationTimeout
 	status := btcDel.GetStatus(btcTipHeight, wValue, params.CovenantQuorum)
-	if status != types.BTCDelegationStatus_PENDING {
-		ms.Logger(ctx).Debug("Received covenant signature after the BTC delegation is already expired", "covenant pk", req.Pk.MarshalHex())
-		return &types.MsgAddCovenantSigsResponse{}, nil
+	if status == types.BTCDelegationStatus_UNBONDED {
```

Comment:
Just note here: I guess this change allow to accept more signatures than the quorum. I imagine slasher or some other tool may relay of there being only quorum of signatures when building witness for sending slashing transactions. This may require some updates there. ( I definitely remember staker, assumes there is only quorum of sigs on Babylon chain)

Reply:
Hmm will check when bumping peripheral repositories. In which situation BTC staker will need to rely on this behaviour? As far as I know in other repositories (eg vigilante) it does not even rely on how many covenant signatures are in a BTC delegation, as slashing only uses one of all covenant signatures. Did I miss anything?

Reply:
What do you mean by `as slashing only uses one of all covenant signatures` ?

The problem I have in mind is for example here: https://github.com/babylonlabs-io/vigilante/blob/main/btcstaking-tracker/btcslasher/slasher_utils.go#L344. 

We build the wittness from all covenant signatures in the delegation. Up until now it was fine, because we accepted only quorum of signatures in the delegation. 

After this change this is not fine now we can have more that quorum signatures in delegation, but BTC script expects the witness to have exactly quorum of signatures.